index.js

```diff
@@ -15,8 +15,8 @@ function plugin (fn, options = {}) {
 
   fn[Symbol.for('skip-override')] = true
 
-  if (typeof fn.options === 'object' && fn.options !== null) {
-    fn[Symbol.for('base-options')] = fn.options
+  if (typeof fn.autoConfig === 'object' && fn.autoConfig !== null) {
```

Comment:
Is `fn.autoConfig = ['foo']` valid?

Reply:
yes, in as much as passing an array as `opts` is also valid - but I'm closing this PR because of https://github.com/fastify/fastify-autoload/pull/60#issuecomment-571596412 - I think we can simplify without doing this in fastify-plugin, while promoting the key use case